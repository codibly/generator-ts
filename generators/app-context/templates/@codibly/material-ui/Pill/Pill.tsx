import styled from '@emotion/styled';
import { Theme as MuiTheme, WithTheme } from '@material-ui/core';
import * as React from 'react';
import { ComponentType, HTMLAttributes } from 'react';

export namespace Pill {
  export type Props = HTMLAttributes<HTMLSpanElement>;
}

function getTypography(theme?: MuiTheme) {
  if (theme && theme.typography) {
    return theme.typography.caption;
  }

  return {};
}

export const Pill = styled.span(({ theme }: Partial<WithTheme>) => ({
  ...getTypography(theme),
  display: 'inline-flex',
  outline: 'none',
  whiteSpace: 'nowrap',
  borderRadius: 16,
  height: 32,
  verticalAlign: 'middle',
  justifyContent: 'center',
  textDecoration: 'none',
  alignItems: 'center',
  boxSizing: 'border-box',
  padding: '0 12px'
})) as ComponentType<Pill.Props>;
