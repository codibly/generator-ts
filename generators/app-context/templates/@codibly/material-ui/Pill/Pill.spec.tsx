import * as React from 'react';
import { render } from '@testing-library/react';
import { renderInApp } from 'App/test/renderInApp';
import { Pill } from './Pill';

describe('Pill', () => {
  it('should render a pill', () => {
    const { getByText, container } = renderInApp(<Pill>Test</Pill>);

    expect(getByText('Test')).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it('should render without theme', () => {
    const { getByText } = render(<Pill>Without theme</Pill>);
    expect(getByText('Without theme')).toBeDefined();
  });
});
