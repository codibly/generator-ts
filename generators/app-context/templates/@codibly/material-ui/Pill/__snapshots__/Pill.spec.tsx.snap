// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill should render a pill 1`] = `
.emotion-0 {
  color: #5F5F5F;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.66;
  -webkit-letter-spacing: 0.03333em;
  -moz-letter-spacing: 0.03333em;
  -ms-letter-spacing: 0.03333em;
  letter-spacing: 0.03333em;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  outline: none;
  white-space: nowrap;
  border-radius: 16px;
  height: 32px;
  vertical-align: middle;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  padding: 0 12px;
}

<div>
  <div />
  <span
    class="emotion-0"
  >
    Test
  </span>
</div>
`;
